examples/js/loaders/ColladaLoader.js

```diff
@@ -3190,6 +3190,14 @@ THREE.ColladaLoader.prototype = {
 			for ( var i = 0, l = keys.length; i < l; i ++ ) {
 
 				var id = instanceMaterials[ keys[ i ] ];
+
+				if ( id === undefined ) {
+
+					console.warn( 'THREE.ColladaLoader: Material with key %s not found. Apply fallback material.', keys[ i ] );
+					id = instanceMaterials[ Object.keys( instanceMaterials )[ 0 ] ];
+
+				}
+
 				materials.push( getMaterial( id ) );
```

Comment:
Um, i wonder if something like this is the more general solution:
```javascript
if ( id === undefined ) {

   console.warn( 'THREE.ColladaLoader: Material with key %s not found. Apply fallback material.', keys[ i ] );
   materials.push( new THREE.MeshPhongMaterial() );

else {

   materials.push( getMaterial( id ) );

}
```
Your approach does only work if there is an entry is `instanceMaterials`. But what about the files where not material definition is found?

Reply:
Yes, it only works if there is at least one ```instance_material```entry...
In my case the problem wasn't that there is no entry but that just the linking was exported wrong but a material was defined. Not quite sure which is better. 
In my approach I more or less "fix" the linking which is pretty easy when there is only one ```instance_material``` defined so the model will look okay. 
In your case the model/part gets some random color... Hmm probably a mix of both? 

Suggestion:
If no linking was found => take the "first" material => if there isn't any material create a random MeshPhongMaterial?

Reply:
I'm not sure if the "take the first material" approach is too use case specific. It might fit for you, but maybe fails in most other cases. Let's see what @mrdoob thinks about that.

>In your case the model/part gets some random color

We could easily define a material outside the for-loop with a fix color that is reused for all missing materials.
```javascript
function resolveMaterialBinding( keys, instanceMaterials ) {

   var materials = [];
   var fallbackMaterial = new THREE.MeshPhongMaterial( { color: 0xffffff ) };

   for ( var i = 0, l = keys.length; i < l; i ++ ) {

      var id = instanceMaterials[ keys[ i ] ];

      if ( id === undefined ) {

         console.warn( 'THREE.ColladaLoader: Material with key %s not found. Apply fallback material.', keys[ i ] );
         materials.push( fallbackMaterial );

      } else {

         materials.push( getMaterial( id ) );

      }

   }

return materials;

}
```

Reply:
This would be my preferred solution right now...

Reply:
Hmm yeah would also be a good idea. Probably take ```0xff00ff``` as preferred color for wrong materials so we make sure that the user notice something is wrong ;) 
But wouldn't it be better if we would only define the material once for the whole loader? I would suggest to name it something like "errorMaterial" and put it in the scope of the Loader. Because every part should use the exact same material anyway (if they are "wrong") so there is no need to create a new one each time we get into ```resolveMaterialBinding```, is there?

Reply:
Yes, we should place the fallback material in the scope of the `parse()` function 👍

Reply:
Can you please change the PR to this approach. I think this is something we can merge.
```javascript
var fallbackMaterial = new THREE.MeshPhongMaterial( { color: 0xff00ff ) };

function resolveMaterialBinding( keys, instanceMaterials ) {

   var materials = [];
   
   for ( var i = 0, l = keys.length; i < l; i ++ ) {

      var id = instanceMaterials[ keys[ i ] ];

      if ( id === undefined ) {

         console.warn( 'THREE.ColladaLoader: Material with key %s not found. Apply fallback material.', keys[ i ] );
         materials.push( fallbackMaterial );

      } else {

         materials.push( getMaterial( id ) );

      }

   }

   return materials;

}

Reply:
sure ;) 
done.